server/schedule/filter/filters.go

```diff
@@ -77,6 +77,9 @@ func Target(opt opt.Options, store *core.StoreInfo, filters []Filter) bool {
 	storeAddress := store.GetAddress()
 	storeID := fmt.Sprintf("%d", store.GetID())
 	for _, filter := range filters {
+		if filter == nil {
```

Comment:
When will this case happen?

Reply:
`NewPlacementLeaderSafeguard` will return nil if placementRule is not enabled.

Reply:
But it will check if `leaderFilter != nil` before appending.